### PR TITLE
feature/jl/fix_values_ws

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java [hunk 1]
```diff
@@ -224,7 +225,8 @@ private void setValue(Setting setting, Settings.Setting.Builder valueBuilder) {
         }
         valueBuilder.setFieldsValues(builder);
       } else if (definition.multiValues()) {
-        valueBuilder.setValues(Settings.Values.newBuilder().addAllValues(COMMA_SPLITTER.split(value)));
+        List<String> values = COMMA_SPLITTER.splitToList(value).stream().map(v -> v.replace(COMA_ENCODED_VALUE, ",")).collect(Collectors.toList());
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java]: `COMA_ENCODED_VALUE` -> `COMMA_ENCODED_VALUE`

**Comment by teryk** [comment]
check feedback and LGTM

File: server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java [hunk 2]
```diff
@@ -165,7 +166,10 @@ private void checkAdminPermission(Optional<ComponentDto> component) {
 
   private Map<String, String> getKeysToDisplayMap(Set<String> keys) {
     return keys.stream()
-      .collect(Collectors.toMap(propertyDefinitions::validKey, Function.identity()));
+      .collect(Collectors.toMap(propertyDefinitions::validKey, Function.identity(),
+        (u, v) -> {
+          throw new IllegalArgumentException(format("'%s' and '%s' cannot be used at the same time as they refer to the same setting", u, v));
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java]: I find it surprising to use the merge function to throw an exception

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java]: In fact I just override what is already done in the default merge function, I just replace the IllegalStateException by an IllegalArgumentException and set a more suitable message.

**Pull request merged**